src/Cello/CMakeLists.txt

```diff
@@ -0,0 +1,99 @@
+include(charm)
+addCharmModule( "mesh" )
+addCharmModule( "main" )
+addCharmModule( "simulation" )
+addCharmModule( "main_enzo" )
+addCharmModule( "main_mesh" )
+addCharmModule( "main_simulation" )
+
+
+function(addCelloLib LIBNAME EXTRASRCS)
+  # Arguments:
+  #   LIBNAME:   Name of the Cello compontent to be build.
+  #   EXTRASRCS: Extra sources to be included to this target.
+  # Add static library based on glob for the library name and including (potentially) excessive dependencies.
+
+  file(GLOB SRC_FILES ${LIBNAME}*.cpp *{LIBNAME}*.hpp)
+  add_library(${LIBNAME} STATIC ${SRC_FILES} ${EXTRASRCS})
+  add_dependencies(${LIBNAME} simulationCharmModule mainCharmModule meshCharmModule main_meshCharmModule main_simulationCharmModule)
+  target_include_directories (${LIBNAME} PRIVATE ${CMAKE_CURRENT_BINARY_DIR} ${CHARM_INCLUDE_DIRS})
+  target_link_options(${LIBNAME} PRIVATE ${Cello_TARGET_LINK_OPTIONS})
+
+endfunction(addCelloLib)
+
+addCelloLib(array "main.cpp")
+target_link_libraries(array simulation)
+addCelloLib(cello   "")
+target_link_libraries(cello HDF5_C)
+addCelloLib(charm   "")
+target_link_libraries(charm HDF5_C)
+addCelloLib(compute "")
+target_link_libraries(compute HDF5_C)
+addCelloLib(control "")
+target_link_libraries(control HDF5_C)
+addCelloLib(disk "main.cpp")
+target_link_libraries(disk error HDF5_C cello)
+addCelloLib(error "main.cpp")
+target_link_libraries(error monitor Boost::filesystem)
+addCelloLib(data  "main_simulation.cpp;field_face_store.F")
+target_link_libraries(data HDF5_C mesh io parallel cello)
+addCelloLib(io    "main_simulation.cpp")
+target_link_libraries(io mesh HDF5_C)
+# need to figure out what to do with enzo from main_enzo in this src tree
+#addCelloLib(main  "")
```

Comment:
Should this comment remain?

Reply:
I left it as skeleton, but will remove it momentarily.